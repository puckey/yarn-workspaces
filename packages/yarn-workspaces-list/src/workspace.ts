export type Name = string;
export type AbsolutePath = string;
export type RelativePath = string;

export interface Workspace {
  name: Name;
  path: AbsolutePath;
  location: RelativePath;
  workspaceDependencies: Name[];
  mismatchedWorkspaceDependencies: Name[];
  transitiveWorkspaceDependencies: Name[];
}
