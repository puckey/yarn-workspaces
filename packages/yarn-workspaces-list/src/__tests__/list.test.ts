import { join, relative } from 'path';
import { listWorkspaces } from '../list';
import { Workspace } from '../workspace';

jest.mock('child_process');

test('should list workspaces', async () => {
  const cwd = join(__dirname, '../../__fixtures__');

  const list = await listWorkspaces({ cwd });

  expect(list.map(normalize(cwd))).toMatchSnapshot();
});

function normalize(cwd: string) {
  return (workspace: Workspace): Workspace => ({
    ...workspace,
    path: `absolute(${relative(cwd, workspace.path)})`,
  });
}
