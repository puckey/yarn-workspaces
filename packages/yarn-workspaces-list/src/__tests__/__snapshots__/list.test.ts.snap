// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should list workspaces 1`] = `
Array [
  Object {
    "location": ".",
    "mismatchedWorkspaceDependencies": Array [],
    "name": "workspaces",
    "path": "absolute()",
    "transitiveWorkspaceDependencies": Array [],
    "workspaceDependencies": Array [
      "a",
    ],
  },
  Object {
    "location": "packages/a",
    "mismatchedWorkspaceDependencies": Array [],
    "name": "a",
    "path": "absolute(packages/a)",
    "transitiveWorkspaceDependencies": Array [],
    "workspaceDependencies": Array [],
  },
  Object {
    "location": "packages/b",
    "mismatchedWorkspaceDependencies": Array [],
    "name": "b",
    "path": "absolute(packages/b)",
    "transitiveWorkspaceDependencies": Array [],
    "workspaceDependencies": Array [
      "a",
    ],
  },
  Object {
    "location": "packages/c",
    "mismatchedWorkspaceDependencies": Array [],
    "name": "c",
    "path": "absolute(packages/c)",
    "transitiveWorkspaceDependencies": Array [],
    "workspaceDependencies": Array [
      "a",
    ],
  },
  Object {
    "location": "projects/d",
    "mismatchedWorkspaceDependencies": Array [],
    "name": "d",
    "path": "absolute(projects/d)",
    "transitiveWorkspaceDependencies": Array [
      "a",
    ],
    "workspaceDependencies": Array [
      "b",
      "c",
    ],
  },
]
`;
